File: spring-mvc/spring-mvc-1/src/main/java/com/springservletthymeleaf/basic/itemservice/domain/item/Item.java
package com.springservletthymeleaf.basic.itemservice.domain.item;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@ToString
@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = false)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Item {

	@Id
	@GeneratedValue
	@Column(name = "item_id")
	@EqualsAndHashCode.Include
	private Long id;

	@EqualsAndHashCode.Include
	private String name;

	@EqualsAndHashCode.Include
	private int price;

	@EqualsAndHashCode.Include
	private int quantity;

	@Builder
	public Item(String name, int price, int quantity) {
		this.name = name;
		this.price = price;
		this.quantity = quantity;
	}
}

File: spring-mvc/spring-thymeleaf/src/main/kotlin/com/springthymeleaf/message/Item.kt
package com.springthymeleaf.message

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity
class Item(
  @Id
  @GeneratedValue
  var id: Long? = null,
  var name: String,
  var price: Int,
  var quantity: Int
) {
  constructor() : this(null, "", 0, 0)

  override fun toString(): String {
    return "Item(id=$id, name='$name', price=$price, quantity=$quantity)"
  }

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false

    other as Item

    if (id != other.id) return false
    return true
  }

  override fun hashCode(): Int {
    return id?.hashCode() ?: 0
  }
}